oio/blob/cache.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2015-2017 OpenIO SAS, as part of OpenIO SDS
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3.0 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.
+
+
+from time import time
+from urlparse import urlparse
+
+from oio.common.http_urllib3 import get_pool_manager
+from oio.common.client import ProxyClient
+from oio.common.logger import get_logger
+
+
+TIMEOUT_CACHE = 60  # in seconds
+
+
+class ServiceCache(object):
+    def __init__(self, conf, pool_manager=None):
+        self._cache = dict()
+        self.conf = conf
+        self.pool_manager = pool_manager or get_pool_manager()
+        self._client = ProxyClient(self.conf, pool_manager=self.pool_manager,
+                                   request_prefix="/conscience")
+        self.logger = get_logger(conf)
+
+    def _get_addr(self, item):
+        if item in self._cache and self._cache[item]['ts'] < time():
+            return self._cache[item]['addr']
+        resp, body = self._client._request(
+            "GET", "/resolve", params={'type': 'rawx', 'uuid': item})
```

Comment:
The host resolving request (without caching) could be implemented in `ConscienceClient`.